addons/website_slides/controllers/main.py

```diff
@@ -787,11 +798,33 @@ def slide_set_completed(self, slide_id):
         fetch_res = self._fetch_slide(slide_id)
         if fetch_res.get('error'):
             return fetch_res
-        self._set_completed_slide(fetch_res['slide'])
+
+        slide = fetch_res['slide']
+        if not slide.channel_id.is_member:
+            return {'channel_completion': 0}
+
+        self._set_completed_slide(slide)
         return {
             'channel_completion': fetch_res['slide'].channel_id.completion
         }
 
+    @http.route('/slides/slide/<model("slide.slide"):slide>/set_uncompleted', website=True, type='http', auth='user')
+    def slide_set_uncompleted_and_reload(self, slide):
+        self._set_uncompleted_slide(slide)
+        return werkzeug.utils.redirect(f'/slides/slide/{slug(slide)}')
+
+    @http.route('/slides/slide/set_uncompleted', website=True, type='json', auth='public')
+    def slide_set_uncompleted(self, slide_id):
+        if request.website.is_public_user():
+            return {'error': 'public_user'}
+        fetch_res = self._fetch_slide(slide_id)
+        if fetch_res.get('error'):
+            return fetch_res
+        self._set_uncompleted_slide(fetch_res['slide'])
+        return {
+            'channel_completion': fetch_res['slide'].channel_id.completion,
+        }
+
```

Comment:
yes, it's called in `slide_course_page.js@_setCompleted`
(you do not see it when you grep because the route is build from a variable)
```
const route = completed ? 'set_completed' : 'set_uncompleted';
const data = await this._rpc({
    route: '/slides/slide/' + route,
    params: {slide_id: slide.id},
});
```